packages/mdc-top-app-bar/short/foundation.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {cssClasses} from '../constants';
+import MDCTopAppBarAdapter from '../adapter';
+import MDCTopAppBarFoundation from '../foundation';
+
+/**
+ * @extends {MDCTopAppBarFoundation<!MDCShortTopAppBarFoundation>}
+ * @final
+ */
+class MDCShortTopAppBarFoundation extends MDCTopAppBarFoundation {
+  /**
+   * @param {!MDCTopAppBarAdapter} adapter
+   */
+  constructor(adapter) {
+    super(adapter);
+    // State variable for the current top app bar state
+    this.isCollapsed = false;
+
+    this.scrollHandler_ = () => this.shortAppBarScrollHandler_();
+  }
+
+  init() {
+    const isAlwaysCollapsed = this.adapter_.hasClass(cssClasses.SHORT_COLLAPSED_CLASS);
```

Comment:
I think you could call super.initi() and then not have to register the scroll handler